src/router/hooks/guards/domainsLimitGuard.js

```diff
@@ -0,0 +1,16 @@
+import { listWorkloadsService } from '@/services/workloads-services'
+/** @type {import('vue-router').NavigationGuardWithThis} */
+
+export async function domainsLimitGuard({ to }) {
+  const workloads = await listWorkloadsService({
+    fields: 'id',
+    ordering: 'id',
+    page: 1,
+    pageSize: 1
+  })
+
+  const isMaxDomainsReached = workloads.count >= 3000
+  if ((to.name === 'create-data-stream' || to.name === 'edit-data-stream') && isMaxDomainsReached) {
```

Comment:
Assim como você fez anteriormente ao criar uma variável para esclarecer o máximo de itens permitidos, seria interessante aplicar a mesma abordagem aqui: `(to.name === 'create-data-stream' || to.name === 'edit-data-stream')`. Isso tornaria o código mais legível e fácil de entender.